### DISCUSSION TITLE
Create a field based on other fields

### DISCUSSION THREAD

**Discussion by tiddle** [pocketbase/pocketbase#5040] [Q&A]
Can i create a field that is based on another field? eg i want a field that takes an input of the `name` and `date` to form a value that is `my_name-2024_01_01` in another field called `slug`. I still want the `slug` field to be editable though.

I have done something like this in contentful but had to write a custom plugin for it.

**Comment by ganigeorgiev** [comment]
No, at least not out of the box.

As a workaround this could be done on the client-side before submitting the value OR on the server-side programmatically via the [Go](https://pocketbase.io/docs/go-event-hooks/) or [JSVM](https://pocketbase.io/docs/js-event-hooks/) hooks.

**Comment by tiddle** [comment]
For anyone following along, here's the code I came up with

Updated Code:
```js
onModelBeforeCreate((e) => {
    const { createSlug } = require(`${__hooks}/util`);
	const name = e.model.get('name');
	const date = e.model.get('startDate');
	const newSlug = createSlug(name, date);

	e.model.set('slug', newSlug);
}, 'events');

onModelBeforeUpdate((e) => {
    const { createSlug } = require(`${__hooks}/util`);

	if (e.model.get('slug')) {
		const name = e.model.get('name');
		const date = e.model.get('startDate');
		const newSlug = createSlug(name, date);

		e.model.set('slug', newSlug);
	}
}, 'events');
```

Old code, **don't use**, kept here for so the replies make sense
```js
const { createSlug } = require('./util');

onRecordBeforeCreateRequest((e) => {
	const name = e.model.get('name');
	const date = e.model.get('startDate');
	const newSlug = createSlug(name, date);

	e.model.set('slug', newSlug);
}, 'events');

onModelBeforeUpdate((e) => {
	if (e.model.get('slug')) {
		const name = e.model.get('name');
		const date = e.model.get('startDate');
		const newSlug = createSlug(name, date);

		e.model.set('slug', newSlug);
	}
}, 'events');
```

**Reply by ganigeorgiev** [reply]
The above is OK but note that the record hooks doesn't have `e.model` but `e.record` (this is normalized with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355)). If you want to keep it with the `e.model` you can replace `onRecordBeforeCreateRequest` with `onModelBeforeCreate`.

Also keep in mind that the `*Request` hooks react **only** on API requests, while the db hooks (`onModel*`) reacts on every db Dao change.

**Reply by ganigeorgiev** [reply]
Also variable and functions outside of the handler scope (in your case `createSlug`) may not be accessible. See https://pocketbase.io/docs/js-overview/#handlers-scope.
